From 69e252868d0046a29aeb91e9f7030f3e795b9db3 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 18 Mar 2021 20:22:05 +0800
Subject: [PATCH 204/204] arm64: allwinner: dts: a64: add AF8133J to PinePhone

New batches of PinePhones switched the magnetometer to AF8133J from
LIS3MDL because lack of ST components.

Add this magnetometer to the device tree. These two sensors do not
conflict with each other, and the non-existent sensor's driver won't
just get probed properly.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
index 4990fd876266..dfd5c5e4e895 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
@@ -325,6 +325,14 @@
 	status = "okay";
 	clock-frequency = <400000>;
 
+	/* Alternative magnetometer */
+	af8133j: magnetometer@1c {
+		compatible = "voltafield,af8133j";
+		reg = <0x1c>;
+		avdd-supply = <&reg_dldo1>;
+		dvdd-supply = <&reg_dldo1>;
+	};
+
 	/* Magnetometer */
 	lis3mdl: lis3mdl@1e {
 		compatible = "st,lis3mdl-magn";
-- 
2.20.1

